node/libs/crypto/src/bn254/mod.rs

```diff
@@ -25,52 +27,66 @@ pub mod hash;
 mod testonly;
 
 /// Type safety wrapper around a scalar value.
+#[derive(Debug, PartialEq)]
 pub struct SecretKey(Fr);
 
 impl SecretKey {
     /// Gets the corresponding [`PublicKey`] for this [`SecretKey`]
     pub fn public(&self) -> PublicKey {
-        let p = G2::generator() * self.0;
+        let p = G2Affine::one().mul(self.0);
         PublicKey(p)
     }
 
     /// Produces a signature using this [`SecretKey`]
     pub fn sign(&self, msg: &[u8]) -> Signature {
         let hash_point = hash::hash_to_g1(msg);
-        let sig = hash_point * self.0;
+        let sig = hash_point.mul(self.0);
         Signature(sig)
     }
 }
 
 impl ByteFmt for SecretKey {
     fn decode(bytes: &[u8]) -> anyhow::Result<Self> {
-        Fr::deserialize_compressed(bytes)
-            .map(Self)
-            .context("failed to decode secret key")
+        let mut fr_repr = FrRepr::default();
+        fr_repr.read_be(Cursor::new(bytes))?;
```

Comment:
- Nit: `Cursor` is not strictly necessary here; `std::io::Read` is implemented for `&mut &[u8]` as well (i.e., a shared reference to bytes, where the reference itself can change). Thus, you can declare `mut bytes: &[u8]` and call `fr_repr.read_be(&mut bytes)` and maybe check that all bytes were read.
- Is big-endian encoding canonical for BN254 (i.e., used in other implementations)? Or is there no convention? I'm not entirely sure, but the `ark` implementation looks little-endian, although [the Go one by Consensys](https://pkg.go.dev/github.com/consensys/gnark-crypto/ecc/bn254) is big-endian.

Reply:
> - Nit: `Cursor` is not strictly necessary here; `std::io::Read` is implemented for `&mut &[u8]` as well (i.e., a shared reference to bytes, where the reference itself can change). Thus, you can declare `mut bytes: &[u8]` and call `fr_repr.read_be(&mut bytes)` and maybe check that all bytes were read.

Right it's not necessary. Simply removing it seems to be enough. 

> - Is big-endian encoding canonical for BN254 (i.e., used in other implementations)? Or is there no convention? I'm not entirely sure, but the `ark` implementation looks little-endian, although [the Go one by Consensys](https://pkg.go.dev/github.com/consensys/gnark-crypto/ecc/bn254) is big-endian.

I'm not sure, but I don't think that there is a convention regarding the endianness.